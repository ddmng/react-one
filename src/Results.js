import React, { Component } from "react";
import "./App.css";
import Pet from "./Pet";

class Results extends Component {
    constructor(props) {
        super(props);

        this.state = {
            name: "",
            list: [{
                name: "fuffi",
                id: 1,
                media: 'media',
                location: 'Italy',
                breed: 'doberman',
                animal: 'dog'
            }, {
                name: "lillo",
                id: 2,
                media: 'media',
                location: 'Italy',
                breed: 'siamese',
                animal: 'cat'
            }]
        };
    }

    addPet = () => {
        console.log("addPet");
        this.setState({
            name: "",
            list: [
                ...this.state.list,
                {
                    name: this.state.name,
                    id: this.state.name
                }
            ]
        });
    };

    updateName = event => {
        console.log("updateName");
        this.setState({
            name: event.target.value
        });
    };

    render() {
        return (
            <div className="pets">
                {this.state.list.map(p => (
                    <Pet name={p.name}
                        key={p.id}
                        id={p.id}
                        animal={p.animal}
                        media={p.media}
                        location={p.location}
                        breed={p.breed} />
                ))}
            </div>
        );
    }
}

export default Results;
