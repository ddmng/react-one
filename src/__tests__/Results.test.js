import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Results from "../Results";

describe('Results', () => {
  const handleDeleteClick = id => () => {
    console.log("clicked");
  };
  const pets = [
    {
      name: "fuffi",
      id: 1,
      media: "media",
      location: "Italy",
      breed: "doberman",
      animal: "dog"
    },
    {
      name: "lillo",
      id: 2,
      media: "media",
      location: "Italy",
      breed: "siamese",
      animal: "cat"
    }
  ];
  const filter = "";


it("renders without crashing", () => {
    const div = document.createElement("div");
  
    ReactDOM.render(
      <Results
        filter={filter}
        pets={pets}
        handleDeleteClick={handleDeleteClick}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  
  });
  
  it('matches the saved snapshot', () => {
  
    const component = renderer.create(
      <Results
        filter={filter}
        pets={pets}
        handleDeleteClick={handleDeleteClick}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
  
})
