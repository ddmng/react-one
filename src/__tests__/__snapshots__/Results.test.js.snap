// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results matches the saved snapshot 1`] = `
<div
  className="pets"
>
  <div>
    <div
      className="pet"
    >
      <a
        href="/details/1"
        onClick={[Function]}
      >
        <h1>
          Nome: 
          fuffi
        </h1>
      </a>
      <h4>
         -  
        dog
      </h4>
      <h4>
         -  
        doberman
      </h4>
      <h4>
         -  
        media
      </h4>
      <h4>
         -  
        Italy
      </h4>
    </div>
    <button
      onClick={[Function]}
    >
      x
    </button>
  </div>
  <div>
    <div
      className="pet"
    >
      <a
        href="/details/2"
        onClick={[Function]}
      >
        <h1>
          Nome: 
          lillo
        </h1>
      </a>
      <h4>
         -  
        cat
      </h4>
      <h4>
         -  
        siamese
      </h4>
      <h4>
         -  
        media
      </h4>
      <h4>
         -  
        Italy
      </h4>
    </div>
    <button
      onClick={[Function]}
    >
      x
    </button>
  </div>
</div>
`;
